internal/metadatastorage/parser.go

```diff
@@ -0,0 +1,26 @@
+package metadatastorage
+
+import (
+	"context"
+
+	"github.com/testifysec/archivista/ent"
+)
+
+var (
+	parsersByPredicate = map[string]ParserFunc{}
+)
+
+type Storer interface {
+	Store(context.Context, *ent.Tx, *ent.Statement) error
```

Comment:
I don't love passing the intoto statement in here.

However, since the predicate is contained directly within the in-toto statement it doesn't seem that bad, since predicates need to make the edge to the statement itself.

Reply:
could we just pass in the statementID?  

ex.
```
func (parsedCollection ParsedCollection) Store(ctx context.Context, tx *ent.Tx, statementID int) error {
    collection, err := tx.AttestationCollection.Create().
        SetStatementID(statementID).
        SetName(parsedCollection.Name).
        Save(ctx)
    if err != nil {
        return err
    }

    for _, a := range parsedCollection.Attestations {
        if err := tx.Attestation.Create().
            SetAttestationCollectionID(collection.ID).
            SetType(a.Type).
            Exec(ctx); err != nil {
            return err
        }
    }

    return nil
}
```

Reply:
We could use an interface{} here, but until we need it it adds complications

Reply:
I'm not sure what you mean by using an interface{} here, and using interface{} or any is a pretty big code smell